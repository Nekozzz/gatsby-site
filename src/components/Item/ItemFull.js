import React from "react";

const ItemFull = ({car}) => {

  return (
    <div key={car.slug}>
      <div>
        {car && (
          <img className={'img-preview'} src={`http://lorempixel.com/460/230/transport/${car.id}/`} alt=""/>
        )}
      </div>
      <div>
        Марка: {car.car_brand.brand_name}
      </div>
      <div>
        Модель: {car.model_name}
      </div>
      <div style={{
        'marginTop': '8px'
      }}>
        Описание:
        <div>
          {car.model_description}
        </div>
      </div>
    </div>
  )
};

export default ItemFull
