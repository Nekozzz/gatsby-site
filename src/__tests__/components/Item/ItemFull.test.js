import React from "react"
import renderer from "react-test-renderer"
import ItemFull from "../../../components/Item/ItemFull"

describe("ItemFull", () => {
  test("renders correctly", () => {
    const tree = renderer
      .create(<ItemFull car={
        {
          car_brand: {
              brand_name: "Ford",
              slug: "ford"
          },
          id: 2,
          model_name: "Ford Focus III",
          model_description: "В сравнении со своим предшественником, Focus-III увеличился в длину на 21 мм",
          slug: "ford-focus-iii",
        }
      }/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})