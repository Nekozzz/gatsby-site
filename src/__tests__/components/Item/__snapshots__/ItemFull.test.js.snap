// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemFull renders correctly 1`] = `
<div>
  asd
  <div
    style={
      Object {
        "marginBottom": "8px",
      }
    }
  >
    <a
      href="/ford"
    >
      Каталог 
      Ford
    </a>
  </div>
  <hr
    style={
      Object {
        "marginBottom": "14px",
      }
    }
  />
  <div>
    <img
      alt=""
      className="img-preview"
      src="http://lorempixel.com/460/230/transport/2/"
    />
  </div>
  <div>
    Марка: 
    Ford
  </div>
  <div>
    Модель: 
    Ford Focus III
  </div>
  <div
    style={
      Object {
        "marginTop": "8px",
      }
    }
  >
    Описание:
    <div>
      В сравнении со своим предшественником, Focus-III увеличился в длину на 21 мм
    </div>
  </div>
</div>
`;
